ideascube/search/tests/test_models.py

```diff
@@ -109,6 +110,23 @@ def test_search_Document_on_tag_name_and_slug():
 
 
 @pytest.mark.usefixtures('cleansearch')
+def test_search_Document_is_case_insensitive():
+    # The custom function to parse tag string (and so sanitize tag's names)
+    # is used is form only, not if we use `tag.set()` or `tag.add()`.
+    # DocumentFactory use `tag.add()` to set the tags of the document and so
+    # the tag's names are not sanitized.
+    doc1 = DocumentFactory(tags=[sanitize_tag_name("aé")])
+    doc2 = DocumentFactory(tags=[sanitize_tag_name("AÉ")])
+    doc3 = DocumentFactory(tags=[sanitize_tag_name("Bar")])
+
+    for d in (doc1, doc2):
+        assert d in Search.search(tags__match=["aé"])
+        assert d in Search.search(tags__match=["AÉ"])
```

Comment:
You're essentially running the search query for each document.

Why not run it once, and then ensure it is **exactly** the list of doc1 and doc2?

That advantageously replaces the above 3 lines, but it also removes the need for the next one.